### PR TITLE
Testnet configuration

### PR DESCRIPTION
**Pull request by eugene-babichenko**
Compose files (`docker-compose.` and `.yml` parts are omitted):

- `dev` - baseline compose file for all operations
- `run` - configuration for running a node in production
- `genesis` - run the first node in a network with the genesis block generated
- `test` - baseline test configuration suitable for PyCharm unittest configuration feature
- `run-test` - configuration for running standalone tests
- `debug-tp` - PyCharm compatible configuration for debugging the transaction processor
- `debug-rest` - PyCharm compatible configuration for debugging the REST API server

Those configurations need to be "layered" e.g. passed to Docker Compose one by one with multiple `-f` keys (see `Makefile` for examples).

Required layers:

- run a node: `dev` + `run`
- run a node in the genesis mode: `dev` + `genesis` + `run`
- debug the transaction processor: `dev` + `debug-tp`
- debug the REST API server: `dev` + `debug-rest`
- run tests in PyCharm: `test`
- run tests in standalone mode (i.e. in a terminal): `test` + `run-test`

Configuration variables are placed in `.env` file.

Volumes used in this configuration:

- `user_keys` - keys used for signing transaction
- `validator_keys` - keys used for submitting new blocks
- `genesis_data` - used for genesis block storage (only in `genesis` configurations)

**TBD in next PRs**

- A separate volume to store chain data

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# The port for communications between validators (blockchain protocol).
+REMME_VALIDATOR_PORT=8800
+# The real IP of the validator (required for validator discovery).
+REMME_VALIDATOR_IP=127.0.0.1
+# Peering mode: in `dynamic` mode validator connects to validators listed in
+# `REMME_SEEDS_LIST` and then starts to connect to other validators dynamically
+# building the network. In `static` mode validator connects only to validators
+# listed in `REMME_PEERS_LIST`.
+REMME_PEERING_MODE=dynamic
+REMME_PEERS_LIST=
```
Comment (bogomazov) [inline .env]: add an example of a value? 
Ex. `REMME_PEERS_LIST=tcp://hostname:port,tcp://hostname:port`

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -19,26 +19,33 @@ services:
...
+          ADDITIONAL_ARGS=\"$$ADDITIONAL_ARGS --seeds $$REMME_SEEDS_LIST\"
         fi
-        sawset genesis -k /root/.sawtooth/keys/key.priv && \
-        sawadm genesis config-genesis.batch /genesis/batch/token-proposal.batch && \
         sawtooth-validator -vv \
-          --endpoint tcp://validator:8800 \
+          --endpoint tcp://$$REMME_VALIDATOR_IP:$$REMME_VALIDATOR_PORT \
           --bind component:tcp://eth0:4004 \
-          --bind network:tcp://eth0:8800
+          --bind network:tcp://eth0:8800 \
```
Comment (bogomazov) [inline docker-compose.dev.yml]: "8800 internal port is static" 
Do we not consider the case of VPS running several validators and TPs at once? 
Will there be an internal `port` conflict with other "instances"?
Or 1 VPS = 1 Validator?

File: .env (hunk 1 above)
Comment (bogomazov) [reply .env]: added

**Comment by bogomazov** [comment]
This description would be nice to be written in internal docs, in addition to "running pycharm tests tutorial".

File: docker-compose.dev.yml (hunk 2 above)
Comment (eugene-babichenko) [reply docker-compose.dev.yml]: 1 VPS = 1 Validator I think. If that changes we will adapt, not a big piece of work.

**Comment by eugene-babichenko** [comment]
Sure, I will transfer that.

File: Makefile [hunk 3]
```diff
@@ -15,13 +15,18 @@
...
+RELEASE_NUMBER ?= $(shell git describe --abbrev=0 --tags)
 
-TOTAL_SUPPLY ?= 10000000000000
-export TOTAL_SUPPLY
+include .env
 
-run_dev:
+.PHONY: release
+
+run_dev_no_genesis:
```
Comment (bogomazov) [inline Makefile]: Production?

File: docker-compose.dev.yml [hunk 4]
```diff
@@ -19,26 +19,33 @@ services:
...
       bash -c "
-        sawadm keygen && \
-        sawtooth keygen key
-        if [ -d "/var/lib/sawtooth" ]; then
-          rm /var/lib/sawtooth/*
+        if [ ! -f /etc/sawtooth/keys/validator.priv ]; then
+          sawadm keygen
+        fi
+        ADDITIONAL_ARGS=\"--peering $$REMME_PEERING_MODE\"
+        if [ ! -z \"$$REMME_PEERS_LIST\" ]; then
```
Comment (bogomazov) [inline docker-compose.dev.yml]: is this readable ADDITIONAL_ARGS  if statement masterpiece really needed?
Can we get aways with just
--seeds $$REMME_PEERS_LIST
--peers $$REMME_PEERS_LIST
in the entrypoint?

File: docker-compose.dev.yml [hunk 5]
```diff
@@ -19,26 +19,33 @@ services:
...
+          ADDITIONAL_ARGS=\"$$ADDITIONAL_ARGS --seeds $$REMME_PEERS_LIST\"
```
Comment (bogomazov) [inline docker-compose.dev.yml]: is this really --seeds?

File: docker-compose.dev.yml (hunk 4 above)
Comment (eugene-babichenko) [reply docker-compose.dev.yml]: Will fix it

File: docker-compose.dev.yml (hunk 5 above)
Comment (eugene-babichenko) [reply docker-compose.dev.yml]: Yeah, thanks.

File: Makefile (hunk 3 above)
Comment (eugene-babichenko) [reply Makefile]: No, for production use we will release containers on Docker Hub and separate releases in zip-files.

**Pull request merged**